src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -95,94 +94,20 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
-                        this.LookupMembersInType(
-                            lookupResult,
-                            cloneReturnType,
-                            propName,
-                            arity: 0,
-                            basesBeingResolved: null,
-                            options: LookupOptions.Default,
-                            originalBinder: this,
-                            diagnose: false,
-                            useSiteDiagnostics: ref useSiteDiagnostics);
-                        // PROTOTYPE: Should handle hiding/overriding and bind like regular accesses
-                        if (lookupResult.IsSingleViable &&
-                            lookupResult.SingleSymbolOrDefault is var sym)
-                        {
-                            switch (sym.Kind)
-                            {
-                                case SymbolKind.Property:
-                                    member = sym;
-                                    // PROTOTYPE: this should check for init-only, but that isn't a separate feature yet
-                                    // It also will not work in metadata.
-                                    if (!(sym is SynthesizedRecordPropertySymbol))
-                                    {
-                                        goto default;
-                                    }
-                                    break;
-
-                                default:
-                                    hasErrors = true;
-                                    diagnostics.Add(
-                                        ErrorCode.ERR_WithMemberIsNotRecordProperty,
-                                        location);
-                                    break;
-                            }
-                        }
-
-                        if (!hasErrors && member is null)
-                        {
-                            hasErrors = true;
-                            Error(
-                                diagnostics,
-                                ErrorCode.ERR_NoSuchMemberOrExtension,
-                                location,
-                                cloneReturnType,
-                                propName);
-                        }
-                    }
-
-                    boundRight = BindValue(assignment.Right, diagnostics, BindValueKind.RValue);
-                    if (!(member is null))
-                    {
-                        boundRight = GenerateConversionForAssignment(
-                            member.GetTypeOrReturnType().Type,
-                            boundRight,
-                            diagnostics);
-                    }
-                    lookupResult.Clear();
-                }
-                args.Add((member, boundRight));
-            }
-
-            lookupResult.Free();
+            // N.B. Since we only don't parse nested initializers in syntax there should be no extra
```

Comment:
I don't know how much we care about syntax trees that are built by hand. Consider doing a syntax check even though the parser doesn't produce such trees?

Reply:
We care 0% about syntax trees that are built by hand.

Reply:
Yeah the binder is only resilient to trees that are produced by the parser. This is why for example our source generator API returns `string` not `SyntaxTree`. We wouldn't trust the return if it was a `SyntaxTree` and would essentially call `Parse(sourceGenerator.Generate().ToString())`. Hence have them return `string` directly because it's more efficient.